### PR TITLE
Add log level support

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Please check if all log levels are appropriate and if my env-reading
stuff can be optimized.

### CODE REVIEW

File: src/compositor/background_surface.c [hunk 1]
```diff
@@ -93,7 +93,7 @@ ws_image_buffer_from_png(
     img.version = PNG_IMAGE_VERSION;
 
     if (!png_image_begin_read_from_file(&img, filename)) {
-        ws_log(NULL, "Could not read file: '%s' error: %s.",
+        ws_log(NULL, LOG_NOTICE, "Could not read file: '%s' error: %s.",
```
Comment (TheNeikos) [inline src/compositor/background_surface.c]: Should be `LOG_ERROR`

File: src/compositor/background_surface.c [hunk 2]
```diff
@@ -105,13 +105,14 @@ ws_image_buffer_from_png(
     buff->stride = PNG_IMAGE_ROW_STRIDE(img);
 
     if (!buff->buffer) {
-        ws_log(NULL, "Could not allocate memory for image: '%s'", filename);
+        ws_log(NULL, LOG_NOTICE, "Could not allocate memory for image: '%s'",
```
Comment (TheNeikos) [inline src/compositor/background_surface.c]: Should be `LOG_ERROR` or worse

File: src/compositor/background_surface.c [hunk 3]
```diff
@@ -105,13 +105,14 @@ ws_image_buffer_from_png(
...
+                filename);
         png_image_free(&img);
         return NULL;
     }
 
     if (!png_image_finish_read(&img, NULL, buff->buffer, 0, NULL)) {
-        ws_log(NULL, "Could not read file: '%s'.", filename);
+        ws_log(NULL, LOG_NOTICE, "Could not read file: '%s'.", filename);
```
Comment (TheNeikos) [inline src/compositor/background_surface.c]: Should be `LOG_ERROR`

File: src/compositor/module.c [hunk 4]
```diff
@@ -155,30 +155,30 @@ ws_compositor_init(void) {
 
     retval = populate_connectors();
     if (retval < 0) {
-        ws_log(&log_ctx, "Populate Connectors failed");
+        ws_log(&log_ctx, LOG_DEBUG, "Populate Connectors failed");
```
Comment (TheNeikos) [inline src/compositor/module.c]: This should be critical. It just doesn't mean that no connectors are found, but that a more low level error has happened.

File: src/compositor/module.c [hunk 5]
```diff
@@ -155,30 +155,30 @@ ws_compositor_init(void) {
...
         return retval;
     }
 
-    ws_log(&log_ctx, "Setting modes");
+    ws_log(&log_ctx, LOG_DEBUG, "Setting modes");
     retval = ws_set_select(&ws_comp_ctx.monitors, NULL, NULL,
             set_monitor_modes, NULL);
     if (retval < 0) {
-        ws_log(&log_ctx, "Setting modes failed");
+        ws_log(&log_ctx, LOG_DEBUG, "Setting modes failed");
```
Comment (TheNeikos) [inline src/compositor/module.c]: Same thing here.

File: src/compositor/monitor.c [hunk 6]
```diff
@@ -156,14 +157,14 @@ ws_monitor_populate_fb(
...
-        ws_log(&log_ctx, "Could not create Framebuffer");
+        ws_log(&log_ctx, LOG_CRIT, "Could not create Framebuffer");
     }
     self->saved_crtc = drmModeGetCrtc(ws_comp_ctx.fb->fd, self->crtc);
 
     int ret = drmModeSetCrtc(ws_comp_ctx.fb->fd, self->crtc, self->buffer->fb,
             0, 0, &self->conn, 1, &self->current_mode->mode);
     if (ret) {
-        ws_log(&log_ctx, "Could not set the CRTC for self %d.",
+        ws_log(&log_ctx, LOG_INFO, "Could not set the CRTC for self %d.",
```
Comment (TheNeikos) [inline src/compositor/monitor.c]: This should be an error

File: src/compositor/background_surface.c (hunk 2 above)
Comment (neithernut) [inline src/compositor/background_surface.c]: `LOG_CRIT`

**Comment by matthiasbeyer** [comment]
Updated with new error levels.

**Comment by neithernut** [comment]
ACK from me.

**Pull request merged**